src/main/java/io/r2dbc/postgresql/PostgresqlReplicationConnection.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.r2dbc.postgresql;
+
+import io.r2dbc.postgresql.api.PostgresqlConnectionMetadata;
+import io.r2dbc.postgresql.client.Client;
+import io.r2dbc.postgresql.message.backend.BackendMessage;
+import io.r2dbc.postgresql.message.backend.EmptyQueryResponse;
+import io.r2dbc.postgresql.message.backend.ErrorResponse;
+import io.r2dbc.postgresql.message.backend.ReadyForQuery;
+import io.r2dbc.postgresql.message.frontend.FrontendMessage;
+import io.r2dbc.postgresql.message.frontend.Query;
+import io.r2dbc.postgresql.replication.LogSequenceNumber;
+import io.r2dbc.postgresql.replication.ReplicationRequest;
+import io.r2dbc.postgresql.replication.ReplicationSlot;
+import io.r2dbc.postgresql.replication.ReplicationSlotRequest;
+import io.r2dbc.postgresql.replication.ReplicationStream;
+import io.r2dbc.postgresql.util.Assert;
+import io.r2dbc.spi.Row;
+import reactor.core.publisher.EmitterProcessor;
+import reactor.core.publisher.Mono;
+
+import java.util.function.Predicate;
+
+import static io.r2dbc.postgresql.util.PredicateUtils.or;
+
+/**
+ * Postgres replication connection.
+ */
+final class PostgresqlReplicationConnection implements io.r2dbc.postgresql.api.PostgresqlReplicationConnection {
+
```

Comment:
Perhaps call it `DefaultPostgresqlReplicationConnection`, so you don't have to constant impl vs. API name collision and have to sprinkle your code with FQDN's.